File: Claude_code/new/files/research_job.dart
class ResearchJob {
  const ResearchJob({
    required this.id,
    required this.topic,
    required this.status,
    required this.progress,
    required this.depthMinutes,
    this.resultReport,
    this.modelOverride,
    this.createdAt,
    this.updatedAt,
  });

  final int id;
  final String topic;
  final String status;
  final int progress;
  final int depthMinutes;
  final String? resultReport;
  final String? modelOverride;
  final String? createdAt;
  final String? updatedAt;

  factory ResearchJob.fromJson(Map<String, dynamic> json) {
    return ResearchJob(
      id: json['id'] as int,
      topic: json['topic'] as String,
      status: json['status'] as String,
      progress: json['progress'] as int? ?? 0,
      depthMinutes: json['depth_minutes'] as int? ?? 5,
      resultReport: json['result_report'] as String?,
      modelOverride: json['model_override'] as String?,
      createdAt: json['created_at'] as String?,
      updatedAt: json['updated_at'] as String?,
    );
  }
}

File: Claude_code/new/files/research_job.py
"""
app/models/research_job.py — SQLAlchemy ORM model for research jobs.
"""

from datetime import datetime

from sqlalchemy import Column, DateTime, Integer, String, Text
from sqlalchemy.sql import func

from backend.app.models.base import Base


class ResearchJob(Base):
    __tablename__ = "research_jobs"

    id = Column(Integer, primary_key=True, index=True)
    topic = Column(String(512), nullable=False)
    depth_minutes = Column(Integer, default=5, nullable=False)
    status = Column(String(50), default="queued", nullable=False)
    progress = Column(Integer, default=0, nullable=False)
    result_report = Column(Text, nullable=True)

    # User-selected Ollama model for this job (None = use server default)
    model_override = Column(String(200), nullable=True)

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
    updated_at = Column(
        DateTime(timezone=True),
        onupdate=func.now(),
        nullable=True,
    )
